src/workerd/io/request-tracker.c++

```diff
@@ -2,36 +2,38 @@
 
 namespace workerd {
 
-RequestTracker::Impl::Impl(Hooks& hooksImpl) : hooks(hooksImpl) {};
-RequestTracker::RequestTracker(Hooks& hooksImpl) : impl(kj::refcounted<Impl>(hooksImpl)) {};
+RequestTracker::RequestTracker(Hooks& hooksImpl) : hooks(hooksImpl) {};
 
-RequestTracker::~RequestTracker() noexcept(false) {
-  // Since we're destroying the RequestTracker, we want to prevent any hooks from running after
-  // this point.
-  impl->hooks = nullptr;
+RequestTracker::~RequestTracker() noexcept(false) {}
```

Comment:
Nit: empty destructor? Can the destructor just be removed?

Reply:
This is actually intention since defaulted dtors are `noexcept(true)`.